crates/astria-conductor/src/executor.rs

```diff
@@ -114,6 +119,13 @@ struct Executor<C> {
     /// so that we can mark the block as final on the execution layer when
     /// we receive a finalized sequencer block.
     sequencer_hash_to_execution_hash: HashMap<Vec<u8>, Vec<u8>>,
+
+    /// most recently executed sequencer block hash
+    // TODO pending (GHI-205): reevaluate this after 205 gets merged
+    last_executed_seq_block_hash: Hash,
+
+    /// block queue for blocks that have been recieved but their parent has not been executed yet
+    block_queue: PriorityQueue<SequencerBlockData, Height>,
```

Comment:
What's the purpose of the priority queue here? If I understand this correctly, the goal is to execute the blocks by their height? Could you use just a `BTreeMap` instead that is self-ordering? Or an `IndexMap` followed by `sort_keys`?

I would prefer if we didn't have to derive/implement Hash on `SequencerBlockData` if there isn't a good reason to.

Reply:
I'll look into swapping out the priority queue for one of those.